rfcs/0106-nix-release-schedule.md

```diff
@@ -0,0 +1,92 @@
+---
+feature: Nix release schedule
+start-date: 2021-09-23
+author: Eelco Dolstra
+co-authors: (find a buddy later to help out with the RFC)
+shepherd-team: (names, to be nominated and accepted by RFC steering committee)
+shepherd-leader: (name to be appointed by RFC steering committee)
+related-issues: N/A
+---
+
+# Summary
+[summary]: #summary
+
+Do a new Nix release every 6 weeks.
+
+# Motivation
+[motivation]: #motivation
+
+The last major Nix release, 2.3, came out in September 2019. Since
+then, there has been a huge amount of work on the Nix master branch,
+including major improvements to the new command line interface and new
+experimental features such as flakes, the content-addressable Nix
+store, and recursive Nix. This forces many users to use unstable Nix
+releases, which is undesirable.
+
+In the future, we should avoid having such long gaps between releases,
+since it’s bad for both contributors and users that there is an
+unbounded amount of time before a new feature shows up in a stable
+release. The thing that has historically caused long gaps between Nix
+releases is new experimental features landing in master that we
+weren’t quite sure about, and doing a new release meant having to
+support these features indefinitely. However, Nix 2.4 introduces an
+experimental-features setting that enables us to add such features in
+a way that they can be changed or removed, while still getting
+feedback from adventurous users. So long as experimental features
+don’t cause breakage in stable features, it’s fine to merge them into
+master and include them in a new release.
```

Comment:
While this is true, a lot of smaller changes don't get an experimental feature, e.g. https://github.com/NixOS/nix/pull/4922. Perhaps there should be a generic experimental feature for enabling unstable minor changes such as this, so that we don't have to add an experimental feature for each?